sdk/core/azure_core/src/http/pager.rs

```diff
@@ -120,6 +120,18 @@ impl<T> PageStream<T> {
             stream: Box::pin(stream),
         }
     }
+
+    /// Creates a [`Pager<T>`] from a raw stream of [`Result<T>`](typespec::Result<T>) values.
+    ///
+    /// This constructor is used when you are implementing a completely custom stream and want to use it as a pager.
+    pub fn from_stream<S>(stream: S) -> Self
```

Comment:
@heaths this was quite useful when I wanted to create a more complicated `PageStream` in order to support our cross-partition queries. It seems like a fairly safe addition to the public API for `PageStream`.